Rx/v2/src/rxcpp/operators/impl-retry-repeat.hpp

```diff
@@ -0,0 +1,163 @@
+#pragma once
+
+/*! \file impl-retry-repeat.hpp
+
+    \brief Implementation commonalities between retry and repeat operators abstracted away from rx-retry.hpp and rx-repeat.hpp files. Should be used only from rx-retry.hpp and rx-repeat.hpp
+    
+*/
+
+#include "../rx-includes.hpp"
+
+namespace rxcpp {
+  namespace operators {
+    namespace detail {
+
+      namespace retry_repeat_common {
+        template<class... AN>
+        struct invalid_arguments {};
+
+        template<class... AN>
+        struct invalid : public rxo::operator_base<invalid_arguments<AN...>> {
+          using type = observable<invalid_arguments<AN...>, invalid<AN...>>;
+        };
+        template<class... AN>
+        using invalid_t = typename invalid<AN...>::type;
```

Comment:
invalid should not be shared. These structs are used to provide the user with better information in compilation errors (the operator and the argument types that failed to resolve to a supported overload of that operator). `retry_repeat_common::invalid` is less helpful than `repeat::invalid`